toffy/normalize_test.py

```diff
@@ -0,0 +1,196 @@
+import json
+import os
+import pytest
+
+import numpy as np
+import pandas as pd
+import tempfile
+
+from pytest_cases import parametrize_with_cases
+
+from ark.utils import test_utils, load_utils
+from toffy import normalize
+import toffy.normalize_test_cases as test_cases
+
+parametrize = pytest.mark.parametrize
+
+
+# TODO: move to toolbox repo once created
+def _make_blank_file(folder, name):
+    with open(os.path.join(folder, name), 'w'):
+        pass
+
+
+@parametrize('obj_func_name, num_params', [('poly_2', 3), ('poly_3', 4), ('poly_4', 5),
+                                           ('poly_5', 6), ('log', 2), ('exp', 4)])
+def test_create_objective_function(obj_func_name, num_params):
+
+    obj_func = normalize.create_objective_function(obj_func_name)
+
+    # number of weights + 1 for x
+    inputs = [1] * (num_params + 1)
+
+    _ = obj_func(*inputs)
+
+
+@parametrize('plot_fit', [True, False])
+@parametrize('obj_func', ['poly_2', 'poly_3', 'poly_4', 'poly_5', 'log', 'exp'])
+def test_fit_calibration_curve(plot_fit, obj_func):
+    x_vals = np.random.rand(15)
+    y_vals = np.random.rand(15)
+    _ = normalize.fit_calibration_curve(x_vals, y_vals, obj_func, plot_fit)
+
+
+@parametrize('obj_func, num_params', [('poly_2', 3), ('poly_3', 4), ('poly_4', 5),
+                                      ('poly_5', 6), ('log', 2), ('exp', 4)])
+def test_create_prediction_function(obj_func, num_params):
+    weights = np.random.rand(num_params)
+    pred_func = normalize.create_prediction_function(obj_func, weights)
+
+    _ = pred_func(np.random.rand(10))
+
+
+def test_combine_run_metrics():
+    with tempfile.TemporaryDirectory() as temp_dir:
+
+        # create full directory of files
+        bins = []
+        metrics = []
+        for i in range(1, 5):
+            bin_name = 'example_{}.bin'.format(i)
+            bins.append(bin_name)
+            metric_name = 'example_metric_{}.csv'.format(i)
+            metric_values = {'column_1': np.random.rand(10),
+                             'column_2': np.random.rand(10),
+                             'column_3': np.random.rand(10)}
+            metrics.append([metric_name, metric_values])
+
+        for bin_file in bins:
+            _make_blank_file(temp_dir, bin_file)
+
+        for metric in metrics:
+            name, values_df = metric[0], pd.DataFrame(metric[1])
+            values_df.to_csv(os.path.join(temp_dir, name), index=False)
```

Comment:
these can all be done in the same loop, no?

Reply:
Yup, forgot I was going to create a `case` for this